### PR TITLE
[Storage] Support MaxPageSize, Include, and Filter for listing encryption scopes 

### PR DESCRIPTION
**Pull request by yifanz0**
## Description
Added 3 parameters to Get-AzStorageEncryptionScope - MaxPageSize, Include, and Filter.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs [hunk 1]
```diff
@@ -68,6 +68,21 @@ public class GetAzureStorageEncryptionScopeCommand : StorageFileBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string EncryptionScopeName { get; set; }
 
+        [Parameter(Mandatory = false,
+            HelpMessage = "The maximum number of encryption scopes that will be included in the list response")]
+        public int? MaxPageSize { get; set; }
```
Comment (blueww) [inline src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs:73]: Add check for [ValidateNotNullOrEmpty]

File: src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs [hunk 2]
```diff
@@ -68,6 +68,21 @@ public class GetAzureStorageEncryptionScopeCommand : StorageFileBaseCmdlet
...
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "The filter of encryption scope name. When specified, only encryption scope names starting with the filter will be listed. eg: startswith(name, <prefix>)")]
```
Comment (blueww) [inline src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs:76]: Might better be:
"The filter of encryption scope name. When specified, only encryption scope names starting with the prefix in the filter will be listed. eg: startswith(name, <prefix>)"

Comment (blueww) [inline src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs:76]: Add check for [ValidateNotNullOrEmpty]

File: src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs [hunk 3]
```diff
@@ -68,6 +68,21 @@ public class GetAzureStorageEncryptionScopeCommand : StorageFileBaseCmdlet
...
+        public string Filter { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "Optional, when specified, will list encryption scopes with the specific state. Defaults to All.")]
+        [ValidateSet(ListEncryptionScopesInclude.All,
+            ListEncryptionScopesInclude.Enabled,
+            ListEncryptionScopesInclude.Disabled)]
+        public string Include { get; set; }
```
Comment (blueww) [inline src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs:84]: Add check for [ValidateNotNullOrEmpty]

File: src/Storage/Storage.Management/help/Get-AzStorageEncryptionScope.md [hunk 4]
```diff
@@ -60,6 +62,38 @@ scope2    Enabled  Microsoft.Storage
...
+----      -----    ------             --------------                                         
+scope1    Enabled  Microsoft.Keyvault https://keyvalutname.vault.azure.net:443/keys/keyname
+scope2    Enabled  Microsoft.Storage
+```
+This command lists all enabled encryption scopes of a Storage account, with a max page size of 10 encryption scopes included in each list response. 
+If there are more than 10 encryption scopes to be listed, the command will still list all the encryption scopes, but with multiple requests sent and responses received.
+
+### Example 4: List all disabled encryption scopes with names starting with "test" of a Storage account 
+```powershell
+Get-AzStorageEncryptionScope -ResourceGroupName "myresourcegroup" -AccountName "mystorageaccount" -Include Disabled -Filter �startswith(name, test)�
```
Comment (blueww) [inline src/Storage/Storage.Management/help/Get-AzStorageEncryptionScope.md:83]: better change “ to "

File: src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs (hunk 1 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs:74]: Done

File: src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs (hunk 2 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs]: Done

File: src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs (hunk 3 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs:84]: Done

File: src/Storage/Storage.Management/help/Get-AzStorageEncryptionScope.md (hunk 4 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/help/Get-AzStorageEncryptionScope.md:83]: Done

File: src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs (hunk 2 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/StorageAccount/GetAzureStorageEncryptionScope.cs]: Done

**Review by blueww — APPROVED** [review]

**Pull request merged**